presto-main/src/main/java/io/prestosql/operator/HashSemiJoinOperator.java

```diff
@@ -184,7 +184,7 @@ public SemiJoinPages(SetSupplier channelSetFuture, int probeJoinChannel, Optiona
             // we know the exact size required for the block
             BlockBuilder blockBuilder = BOOLEAN.createFixedSizeBlockBuilder(inputPage.getPositionCount());
 
-            Page probeJoinPage = new Page(inputPage.getBlock(probeJoinChannel));
+            Page probeJoinPage = inputPage.getColumns(probeJoinChannel);
```

Comment:
Maybe that should just be called `getColumn` instead?

Reply:
I thought about that but wanted to overload the varargs method so that single column extraction didn’t accidentally incur the temporary array cost. I could also add a getColumn version of the same thing but would still advocate for the single int overload to make sure accidental usage of the varargs version with a single argument doesn’t incur the temporary array cost. Thoughts?

Reply:
Yes. That makes sense. Thank you.